### PR TITLE
CRNS-133/Attachment

### PR DESCRIPTION
**Pull request by DanC5**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/Attachment/Card.tsx [hunk 1]
```diff
@@ -67,12 +71,38 @@ const goToURL = (url) => {
   });
 };
 
+export type CardProps<
+  At extends UnknownType = DefaultAttachmentType
+> = Attachment<At> & {
+  /**
+   * Position of the message, either 'right' or 'left'
+   */
+  alignment: 'right' | 'left';
```
Comment (virdesai) [inline src/components/Attachment/Card.tsx:80]: we have this type within MessagesContext

File: src/components/Attachment/FileAttachment.tsx [hunk 2]
```diff
@@ -51,83 +55,67 @@ const goToURL = (url) => {
...
-const FileAttachment = ({
+export type FileAttachmentProps<
+  At extends UnknownType = DefaultAttachmentType
+> = {
+  /** Handler for actions. Actions in combination with attachments can be used to build [commands](https://getstream.io/chat/docs/#channel_commands). */
+  actionHandler: ActionHandler;
+  /**
+   * Position of the message, either 'right' or 'left'
+   */
+  alignment: 'right' | 'left';
```
Comment (virdesai) [inline src/components/Attachment/FileAttachment.tsx:66]: we have this type in MessagesContext

File: src/components/Attachment/FileAttachmentGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  align-items: stretch;
+`;
+
+export type FileAttachmentGroupProps<
+  At extends UnknownType = DefaultAttachmentType
+> = {
+  /**
+   * Position of the message, either 'right' or 'left'
+   */
+  alignment: 'right' | 'left';
```
Comment (virdesai) [inline src/components/Attachment/FileAttachmentGroup.tsx:24]: we have this type in MessagesContext

File: src/components/Attachment/Gallery.tsx [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+  GestureResponderEvent,
+  Image,
+  Modal,
+  SafeAreaView,
+  Text,
+  View,
+} from 'react-native';
 import ImageViewer from 'react-native-image-zoom-viewer';
-import styled from 'styled-components/native';
+import type { IImageInfo } from 'react-native-image-zoom-viewer/built/image-viewer.type';
```
Comment (virdesai) [inline src/components/Attachment/Gallery.tsx:11]:
```suggestion

import type { IImageInfo } from 'react-native-image-zoom-viewer/built/image-viewer.type';
```

File: src/components/Attachment/Gallery.tsx [hunk 5]
```diff
@@ -1,16 +1,27 @@
...
+import { useMessageContentContext } from '../../contexts/messageContentContext/MessageContentContext';
+import { useTranslationContext } from '../../contexts/translationContext/TranslationContext';
+import { styled } from '../../styles/styledComponents';
 import { themed } from '../../styles/theme';
 import { makeImageCompatibleUrl } from '../../utils/utils';
 
-const Single = styled.TouchableOpacity`
+import type { DefaultAttachmentType } from '../../types/types';
+
+const Single = styled.TouchableOpacity<{ alignment: 'right' | 'left' }>`
```
Comment (virdesai) [inline src/components/Attachment/Gallery.tsx:24]: we have this type in MessagesContext

File: src/components/Attachment/Gallery.tsx [hunk 6]
```diff
@@ -23,7 +34,10 @@ const Single = styled.TouchableOpacity`
   ${({ theme }) => theme.message.gallery.single.css}
 `;
 
-const GalleryContainer = styled.View`
+const GalleryContainer = styled.View<{
+  alignment: 'right' | 'left';
```
Comment (virdesai) [inline src/components/Attachment/Gallery.tsx:38]: we have this type in MessagesContext

File: src/components/Attachment/Gallery.tsx [hunk 7]
```diff
@@ -72,33 +86,61 @@ const HeaderButton = styled.TouchableOpacity`
...
       <CloseButton />
     </HeaderButton>
   </HeaderContainer>
 );
 
+export type GalleryProps<At extends UnknownType = DefaultAttachmentType> = {
+  /**
+   * Position of the message, either 'right' or 'left'
+   */
+  alignment: 'right' | 'left';
```
Comment (virdesai) [inline src/components/Attachment/Gallery.tsx:105]: we have this type in MessagesContext

File: src/components/Attachment/Attachment.tsx [hunk 8]
```diff
@@ -1,41 +1,132 @@
-import React, { useContext } from 'react';
+import React from 'react';
 import { View } from 'react-native';
-import PropTypes from 'prop-types';
+import type { Attachment as AttachmentType, UnknownType } from 'stream-chat';
```
Comment (virdesai) [inline src/components/Attachment/Attachment.tsx:3]:
```suggestion

import type { Attachment as AttachmentType, UnknownType } from 'stream-chat';
```

File: src/components/Attachment/Attachment.tsx [hunk 9]
```diff
@@ -1,41 +1,132 @@
...
+    groupStyle,
   } = props;
 
   if (!attachment) {
     return null;
   }
 
-  const Giphy = props && props.Giphy ? props.Giphy : Card;
-  const UrlPreview = props && props.UrlPreview ? props.UrlPreview : Card;
+  const Giphy = props?.Giphy ? props.Giphy : Card;
```
Comment (virdesai) [inline src/components/Attachment/Attachment.tsx:128]:
```suggestion
  const Giphy = props?.Giphy || Card;
```

File: src/components/Attachment/Attachment.tsx [hunk 10]
```diff
@@ -1,41 +1,132 @@
...
+  const UrlPreview = props?.UrlPreview ? props.UrlPreview : Card;
```
Comment (virdesai) [inline src/components/Attachment/Attachment.tsx:129]:
```suggestion
  const UrlPreview = props?.UrlPreview || Card;
```

File: src/components/Attachment/Card.tsx (hunk 1 above)
Comment (DanC5) [reply src/components/Attachment/Card.tsx:80]: gotcha, will replace with the Alignment type 👍🏻

**Review by nhannah — APPROVED** [review]

**Pull request merged**